.projenrc.js

```diff
@@ -42,14 +48,13 @@ const project = new cdk.JsiiProject({
   },
   autoApproveUpgrades: true,
 
-  // run upgrade-dependencies workflow at a different hour than other cdk8s
-  // repos to decrease flakiness of integration tests caused by new versions of
-  // cdk8s being published to different languages at the same time
   depsUpgradeOptions: {
     workflowOptions: {
-      schedule: javascript.UpgradeDependenciesSchedule.expressions(['0 1 * * *']),
+      schedule: Cdk8sCommon.upgradeScheduleFor('cdk8s-grafana'),
     },
   },
 });
 
+new Cdk8sCommon(project);
```

Comment:
Very cool! 

The only thing though is that this doesn't allow for common configuration of constructor properties. And since not all components can be configured post instantiation (maybe they should?) - there's a limitation on the capabilities of this approach. 

For example, all our repos define:

```js
autoApproveOptions: {
    allowedUsernames: ['cdk8s-automation'],
    secret: 'GITHUB_TOKEN',
},
autoApproveUpgrades: true,
```

I'd like for this conf to only appear in the common module, and then here do something like:

```js
const project = new cdk.JsiiProject({
  ...Cdk8sCommon.props,
  ...,
})
```

Is this something you had in mind for a later time?

Reply:
Yes! I think that would be pretty straightforward to add. I'll update this with your suggestion.

> The only thing though is that this doesn't allow for common configuration of constructor properties. And since not all components can be configured post instantiation (maybe they should?) - there's a limitation on the capabilities of this approach.

I agree, it does make things a little tricky. Another alternative is to create a custom project type, but I'd prefer against it since it adds another layer of indirection.